graph-courses.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+
+import argparse
+import collections
+import operator
+import calendar as cal
+from Class import *
+from ConvertParse import *
+
+reload(sys)
+sys.setdefaultencoding('utf-8')
+
+args = argparse.ArgumentParser()
+args.add_argument('filename', nargs='+')
+args = args.parse_args()
+courses = collections.defaultdict(list)
+list_course = collections.defaultdict(dict)
+dict_gen = collections.defaultdict(dict)
+dict_view_gen = collections.defaultdict(dict)
+a=0
+
+
+for filename in args.filename:
+	p1 = re.compile(r"(\d+\b)(?!.*\1\b)")
+	filen2 = int(''.join(p1.findall(str(filename))))
+	html = filename.endswith('.htm') or filename.endswith('.html')
+	parse = cal.parseHTML if html else cal.parseText
+	for (name, c) in parse(open(filename)).items():
+		codex = convert_parse_course_id(name, c)
+		course = obscure_refference(name, c)
+		courses[filen2].append(course)
+
```

Comment:
For example, `list_course` could go here (and be renamed to `course_years` or somesuch).